test/default-values.js

```diff
@@ -0,0 +1,177 @@
+                               emptyStringArr: [],
+                               fullStringArr: ['a', 'b'],
+                               emptyBoolArr: [],
+                               fullBoolArr: [false, true, false] },
+                     positionals: [] };
+  const result = parseArgs({ args, options });
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+test('when default is set, the option must be added as result', (t) => {
+  const args = [];
+  const options = {
+    a: { type: 'string', default: 'HELLO' },
+    b: { type: 'boolean', default: false },
+    c: { type: 'boolean', default: true }
+  };
+  const expected = { values: { __proto__: null, a: 'HELLO', b: false, c: true }, positionals: [] };
+
+  const result = parseArgs({ args, options });
+
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+test('when default is set, the args value takes precedence', (t) => {
+  const args = ['--a', 'WORLD', '--b', '-c'];
+  const options = {
+    a: { type: 'string', default: 'HELLO' },
+    b: { type: 'boolean', default: false },
+    c: { type: 'boolean', default: true }
+  };
+  const expected = { values: { __proto__: null, a: 'WORLD', b: true, c: true }, positionals: [] };
+
+  const result = parseArgs({ args, options });
+
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+test('tokens should not include the default options', (t) => {
+  const args = [];
+  const options = {
+    a: { type: 'string', default: 'HELLO' },
+    b: { type: 'boolean', default: false },
+    c: { type: 'boolean', default: true }
+  };
+
+  const expectedTokens = [];
+
+  const { tokens } = parseArgs({ args, options, tokens: true });
+  t.deepEqual(tokens, expectedTokens);
+  t.end();
+});
+
+test('tokens:true should not include the default options after the args input', (t) => {
+  const args = ['--z', 'zero', 'positional-item'];
+  const options = {
+    z: { type: 'string' },
+    a: { type: 'string', default: 'HELLO' },
+    b: { type: 'boolean', default: false },
+    c: { type: 'boolean', default: true }
+  };
+
+  const expectedTokens = [
+    { kind: 'option', name: 'z', rawName: '--z', index: 0, value: 'zero', inlineValue: false },
+    { kind: 'positional', index: 2, value: 'positional-item' },
+  ];
+
+  const { tokens } = parseArgs({ args, options, tokens: true, allowPositionals: true });
+  t.deepEqual(tokens, expectedTokens);
+  t.end();
+});
+
+test('proto as default value must be ignored', (t) => {
+  const args = [];
+  const options = Object.create(null);
+
+  // eslint-disable-next-line no-proto
+  options.__proto__ = { type: 'string', default: 'HELLO' };
+
+  const result = parseArgs({ args, options, allowPositionals: true });
+  const expected = { values: { __proto__: null }, positionals: [] };
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+
+test('multiple as false should expect a String and not an array', (t) => {
+  const args = [];
+  const options = { alpha: { type: 'string', multiple: false, default: 42 } };
```

Comment:
If wish to test array is rejected:
```suggestion
  const options = { alpha: { type: 'string', multiple: false, default: ['string'] } };
```